docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,143 @@
+  * Better to have an explicit request/response from some inventory owner (e.g. the tegde_agent) to the plugin, 
+    to register a new **capability** object for a certain device?
+* Each capability object `tedge_config` contains the field `files`, that tells the plugin the list of config-file the corresponding device provides.
+
+**thin-edge** has a set of pre-defined **capability types**. 
+Each plugin can define plugin-specific capability types, or use one of the pre-defined capability type.
+For details about all [Pre-defined Capability Types](#pre-defined-capability-types) see section below.
+
+### Pre-defined Capability Types
+
+That section lists the pre-defined **capabilities types**.
+
+* Capability Type: **Configration Management**
+
+  |                      |                     | 
+  |:---------------------|:--------------------|
+  | **Unique name**      | `tedge_config` |
+  | **Field:**`files`    | List of config-file the device provides |
+  | **Behavoiur**        | On cloud request provided configuration files are requested from the device and sent to cloud; or downloaded from cloud and sent to the device. For details see [Configuration Managenement documentation](../references/c8y-configuration-management.md#configuration-files-for-child-devices)
+
+* Capability Type: **Logging Management**
+
+  |                      |              | 
+  |:---------------------|:-------------|
+  | **Unique name**      | `tedge_log`  |
+  | **Field:** `files`   | List of log-files the device provides |
+  | **Behavoiur**        | One cloud request provided log-files are requested from the device and sent to cloud. Details see TODO: add link to log plugin documentation, section for child-devices.
+
+The following JSON code shows a small example of an **capability** objects in a device object:
+         
+          {
+              "name": "child-device 1",
+              "type": "thin-edge.io-child",
+              "capabilities": {
+                  "tedge_config": {
+                      files: [ "foo.conf", "bar.conf" ]
+                  },
+                  "tedge_log": {
+                      // ...
+                  },
+              }
+          }
+
+## Registration of a new device
+
+The sequence diagram below illustrates the data/message flow and all components involved, when a new external child-device registers it-self to thin-edge.
+
+```mermaid
+sequenceDiagram
+    participant external child device
+    participant tedge agent
+    participant C8Y cfg plugin
+    participant C8Y log plugin
+    participant mapper
+    participant tedge supported operations API
+    participant C8Y cloud
+        external child device->>tedge agent: (1) register(childid, cfg_capability, log_capability)
+        tedge agent->>mapper: (2) create child-device twin(childid)
+        mapper->>C8Y cloud: (3) create child-device twin(childid)
+        tedge agent->>tedge agent: (4) create child-device twin in the inventory
```

Comment:
I don't understand why the inventory is updated after doing all the updates in the cloud. Shouldn't this be the first thing that the agent does before performing any cloud specific actions? Since the cloud connectivity is not always guaranteed, it'd be better to update the local inventory first and then sync with the cloud as and when it's available.